src/session_impl.cpp

```diff
@@ -630,10 +630,26 @@ namespace aux {
 		update_upnp();
 		update_natpmp();
 		update_lsd();
-		update_dht();
 		update_peer_fingerprint();
-		update_dht_bootstrap_nodes();
-		update_dht_announce_interval();
+
+#ifndef TORRENT_DISABLE_DHT
+		// setup DHT
+		if (m_outstanding_router_lookups == 0)
```

Comment:
If the value of `settings_pack::dht_bootstrap_nodes` is not the default, then `update_dht_bootstrap_nodes` is called and makes `m_outstanding_router_lookups > 0` and when the resolve is done, `update_dht` is called.